python/paddle/trainer/PyDataProvider2.py

```diff
@@ -340,6 +340,9 @@ def __init__(self, file_list, **kwargs):
                 assert self.slots is not None
                 assert self.generator is not None
 
+                if use_dynamic_order and isinstance(self.slots, dict):  # reorder input_types
```

Comment:
If slots is not a dictionary, then should NOT use dynamic order. Also add a function inputs() to specify input order explictly.